pulsar-broker/src/test/java/org/apache/pulsar/broker/stats/PrometheusMetricsTest.java

```diff
@@ -263,33 +273,42 @@ public void testBrokerMetrics() throws Exception {
 
         List<Metric> bytesOutTotal = (List<Metric>) metrics.get("pulsar_broker_out_bytes_total");
         List<Metric> bytesInTotal = (List<Metric>) metrics.get("pulsar_broker_in_bytes_total");
+        List<Metric> topicLevelBytesOutTotal = (List<Metric>) metrics.get("pulsar_out_bytes_total");
+
         assertEquals(bytesOutTotal.size(), 2);
         assertEquals(bytesInTotal.size(), 2);
+        assertEquals(topicLevelBytesOutTotal.size(), 3);
 
         double systemOutBytes = 0.0;
         double userOutBytes = 0.0;
-        switch (bytesOutTotal.get(0).tags.get("system_subscription").toString()) {
-            case "true":
-                systemOutBytes = bytesOutTotal.get(0).value;
-                userOutBytes = bytesOutTotal.get(1).value;
-            case "false":
-                systemOutBytes = bytesOutTotal.get(1).value;
-                userOutBytes = bytesOutTotal.get(0).value;
-        }
-
         double systemInBytes = 0.0;
         double userInBytes = 0.0;
-        switch (bytesInTotal.get(0).tags.get("system_topic").toString()) {
-            case "true":
-                systemInBytes = bytesInTotal.get(0).value;
-                userInBytes = bytesInTotal.get(1).value;
-            case "false":
-                systemInBytes = bytesInTotal.get(1).value;
-                userInBytes = bytesInTotal.get(0).value;
+
+        for (Metric metric : bytesOutTotal) {
+            if (metric.tags.get("system_subscription").equals("true")) {
+                systemOutBytes = metric.value;
+            } else {
+                userOutBytes = metric.value;
+            }
+        }
+
+        for (Metric metric : bytesInTotal) {
+            if (metric.tags.get("system_topic").equals("true")) {
+                systemInBytes = metric.value;
+            } else {
+                userInBytes = metric.value;
+            }
+        }
+
+        double systemCursorOutBytes = 0.0;
+        for (Metric metric : topicLevelBytesOutTotal) {
+            if (metric.tags.get("subscription").startsWith(SystemTopicNames.SYSTEM_READER_PREFIX)) {
```

Comment:
The subscription name may be `__compaction`, we should also consider it.

Reply:
Good point. Addressed